crates/chain/tests/common/tx_template.rs

```diff
@@ -0,0 +1,141 @@
+use std::collections::HashMap;
+
+use bdk_chain::{tx_graph::TxGraph, ConfirmationHeightAnchor, SpkTxOutIndex};
+use bitcoin::{
+    hashes::Hash, locktime::absolute::LockTime, secp256k1::Secp256k1, OutPoint, ScriptBuf,
+    Sequence, Transaction, TxIn, TxOut, Txid, Witness,
+};
+use miniscript::Descriptor;
+
+/// Transaction template.
+pub struct TxTemplate<'a, A> {
+    /// Uniquely identifies the transaction, before it can have a txid.
+    pub tx_name: &'a str,
+    pub inputs: &'a [TxInTemplate<'a>],
+    pub outputs: &'a [TxOutTemplate],
+    pub anchors: &'a [A],
+    pub last_seen: Option<u64>,
+}
+
+impl<'a, A> Default for TxTemplate<'a, A> {
+    fn default() -> Self {
+        Self {
+            tx_name: Default::default(),
+            inputs: Default::default(),
+            outputs: Default::default(),
+            anchors: Default::default(),
+            last_seen: Default::default(),
+        }
+    }
+}
+
+impl<'a, A> Copy for TxTemplate<'a, A> {}
+
+impl<'a, A> Clone for TxTemplate<'a, A> {
+    fn clone(&self) -> Self {
+        *self
+    }
+}
+
+#[allow(dead_code)]
+pub enum TxInTemplate<'a> {
+    /// This will give a random txid and vout.
+    Bogus,
+
+    /// This is used for coinbase transactions because they do not have previous outputs.
+    Coinbase,
+
+    /// Contains the `tx_name` and `vout` that we are spending. The rule is that we must only spend
+    /// a previous transaction.
+    PrevTx(&'a str, usize),
+}
+
+pub struct TxOutTemplate {
+    pub value: u64,
+    pub spk_index: Option<u32>, // some = get spk from SpkTxOutIndex, none = random spk
+}
+
+#[allow(dead_code)]
+pub fn init_graph<'a>(
+    tx_templates: impl IntoIterator<Item = &'a TxTemplate<'a, ConfirmationHeightAnchor>>,
+) -> (
+    TxGraph<ConfirmationHeightAnchor>,
+    SpkTxOutIndex<u32>,
+    HashMap<&'a str, Txid>,
+) {
+    let (descriptor, _) = Descriptor::parse_descriptor(&Secp256k1::signing_only(), "tr(tprv8ZgxMBicQKsPd3krDUsBAmtnRsK3rb8u5yi1zhQgMhF1tR8MW7xfE4rnrbbsrbPR52e7rKapu6ztw1jXveJSCGHEriUGZV7mCe88duLp5pj/86'/1'/0'/0/*)").unwrap();
+    let mut graph = TxGraph::<ConfirmationHeightAnchor>::default();
+    let mut spk_index = SpkTxOutIndex::default();
+    (0..10).for_each(|index| {
+        spk_index.insert_spk(
+            index,
+            descriptor
+                .at_derivation_index(index)
+                .unwrap()
+                .script_pubkey(),
+        );
+    });
+    let mut tx_ids = HashMap::<&'a str, Txid>::new();
+
+    for (bogus_txin_vout, tx_tmp) in tx_templates.into_iter().enumerate() {
+        let tx = Transaction {
+            version: 0,
+            lock_time: LockTime::ZERO,
+            input: tx_tmp
+                .inputs
+                .iter()
+                .map(|input| match input {
+                    TxInTemplate::Bogus => TxIn {
+                        // #TODO have actual random data
```

Comment:
Let's complete this!

A simple solution is to use the current unix timestamp as the seed (which you can hash/modulo).

Also add some bogus inputs to test transactions.